nf_core/pipeline-template/.github/workflows/download_pipeline.yml

```diff
@@ -0,0 +1,68 @@
+name: Test successful pipeline download with 'nf-core download'
+
+# Run the workflow when:
+#  - dispatched manually
+#  - when a PR is opened or reopened to master branch
+#  - the head branch of the pull request is updated, i.e. if fixes for a release are pushed last minute to dev.
+
+on:
+  workflow_dispatch:
+  pull_request:
+    types:
+      - opened
+    branches:
+      - master
+  pull_request_target:
+    branches:
+      - master
+
+env:
+  NXF_ANSI_LOG: false
+
+jobs:
+  download:
+    runs-on: ["self-hosted"]
+    steps:
+      - name: Install Nextflow
+        uses: nf-core/setup-nextflow@v1
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+          architecture: "x64"
+      - uses: eWaterCycle/setup-singularity@v7
+        with:
+          singularity-version: 3.8.3
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install git+https://github.com/nf-core/tools.git@dev
+
+      - name: Get the repository name and current branch set as environment variable
+        run: |
+          echo "REPO_LOWERCASE=${GITHUB_REPOSITORY,,}" >> ${GITHUB_ENV}
+          echo "REPOTITLE_LOWERCASE=$(basename ${GITHUB_REPOSITORY,,})" >> ${GITHUB_ENV}
+          echo "REPO_BRANCH=${GITHUB_REF#refs/heads/}" >> ${GITHUB_ENV}
+
+      - name: Download the pipeline
+        env:
+          NXF_SINGULARITY_CACHEDIR: ./
+        run: |
+          nf-core download ${{ env.REPO_LOWERCASE }} \
+          --revision ${{ env.REPO_BRANCH }} \
+          --outdir ./${{ env.REPOTITLE_LOWERCASE }} \
+          --compress "none" \
+          --container-system 'singularity' \
+          --container-library "quay.io" -l "docker.io" -l "ghcr.io" \
+          --container-cache-utilisation 'amend' \
+          --download-configuration
+
+      - name: Inspect download
+        run: tree ./${{ env.REPOTITLE_LOWERCASE }}
+
+      - name: Run the downloaded pipeline
+        env:
+          NXF_SINGULARITY_CACHEDIR: ./
+          NXF_SINGULARITY_HOME_MOUNT: true
+        run: nextflow run ./${{ env.REPOTITLE_LOWERCASE }}/$( sed 's/\W/_/g' <<< ${{ env.REPO_BRANCH }}) -stub -profile test,singularity --outdir ./results
```

Comment:
What's to stop Nextflow from downloading any missing singularity containers during run time here? eg. if the singularity image has been downloaded with an incorrect filename?

Tagging @mirpedrol @mashehu

Reply:
That is a valid point, although the case of an incorrect filename should be caught by the respective tests in the _tools_ repo. 

Do you think that setting `NXF_OFFLINE` would help? Or should we count the files in the `NXF_SINGULARITY_CACHEDIR` before and after the pipeline run to corroborate that they are still the name number? 

Generally, though, I am already very happy that there is now a download test in the first place, because we had multiple new releases for which `nf-core download` unexpectedly raised exceptions because some odd declaration slipped through the module linting and review. (Such things like arbitrarily mixing single and double quotes). If we can avoid publishing some crooked module as part of a pipeline release like this, I am already very happy!

Reply:
> That is a valid point, although the case of an incorrect filename should be caught by the respective tests in the _tools_ repo.

Good point 👍🏻 



> Do you think that setting `NXF_OFFLINE` would help?

We were chatting about that in the meeting where I added the comment. I'm not confident that it would 👀 😅 We were playing with the idea of cutting off networking access on the custom runner and stuff, but I think it's almost certainly overkill.

No need to do anything here, as you say - invalid container names is something that we should be testing in other tests. Testing for syntax errors / weirdness here is a good addition as it is 👍🏻

Reply:
If you cut networking access entirely, then the most likely test failure will be some issue with the plugins rather than the actual test subject, namely container images. 😅 

I [already searched a bit and there seems to be ways to block network traffic selectively](https://github.com/step-security/github-actions-goat/blob/main/docs/Solutions/RestrictOutboundTraffic.md#network-filtering-with-harden-runner-2), but only on a per job and not on a per-step basis. So if you disallow the registries, the containers would not download successfully beforehand either.